_schemaV2.graphql

```diff
@@ -19427,8 +19464,17 @@ type Viewer {
 
     # Indicates an active subscription
     eligibleForListing: Boolean
+
+    # Exclude partners the user follows (only effective when `include_partners_with_followed_artists` is set to true).
+    excludeFollowedPartners: Boolean
     first: Int
     ids: [String]
+
+    # If true, will only return partners that list artists that the user follows
+    includePartnersWithFollowedArtists: Boolean
+
+    # An IP address, will be used to lookup location
+    ip: String
```

Comment:
**issue:** Would it be possible to use a boolean here to infer that we would like to use the IP for location and infer the location from the request headers. maybe @mzikherman has an idea how we can get it here

Reply:
Yes, Metaphysics can of course accept the ip address, but it should be able to determine it from the request. However, would need to experiment to make sure that works.

I think the reason I suggested passing in the IP address is b/c Force already has some middleware that determines the request's IP - https://github.com/artsy/force/blob/5dc35e740a4a77283b5cd907e9823b688f3af62a/src/Server/passport/lib/app/forwarded_for.js - so I thought it would be simple to do that in Eigen too and pass the IP address along.

I'm surprised Eigen _can't_ do that and requires an external API call in order to find out the IP?

Of course if that's the case, we can move the logic to determine the request's IP to Metaphysics, using that Force middleware as a reference.

Reply:
Added it to MP in https://github.com/artsy/metaphysics/pull/5055